### ISSUE TITLE
OperatorBufferWithTime error

### ISSUE THREAD

**Issue by mariusznet** [ReactiveX/RxJava#3560]
Hi,

According to the java docs 
> When the source {@link Observable} completes or produces an error, the current buffer is emitted, and the event is propagated to all subscribed {@link Subscriber}s."

But in the code on error method clears chunks swallowing them without emitting:
```
        @Override
        public void onError(Throwable e) {
            synchronized (this) {
                if (done) {
                    return;
                }
                done = true;
                chunks.clear();
            }
            child.onError(e);
            unsubscribe();
        }
```

According to the documentation http://reactivex.io/documentation/operators/buffer.html
this behaviour is expected, so the java docs need correction.

**Comment by akarnokd** [comment]
The documentation is wrong. It might be worth extending the API with the usual delayError in 2.x though.

**Comment by akarnokd** [comment]
Fix via #3561.
